tests/baselines/reference/sourceMapValidationDestructuringForOfObjectBindingPatternDefaultValues.symbols

```diff
@@ -93,7 +93,7 @@ for (let {name: nameA = "noName" } of getRobots()) {
 >nameA : Symbol(nameA, Decl(sourceMapValidationDestructuringForOfObjectBindingPatternDefaultValues.ts, 31, 10))
 }
 for (let {name: nameA = "noName" } of [{ name: "mower", skill: "mowing" }, { name: "trimmer", skill: "trimming" }]) {
->name : Symbol(name, Decl(sourceMapValidationDestructuringForOfObjectBindingPatternDefaultValues.ts, 34, 40))

```

Comment:
looks like isInferredTypeOfVariable misses the for-of case as well.

Reply:
Yeah, but again, not sure how much it matters.

Reply:
It probably matters less than the argument case, but it's the same kind of bug in a large code base.

Reply:
it matters for goto def usually. it is a bit disorienting to jump to hit F12 and go somewhere unrelated but similar.

Reply:
also a bit unexpected for renames.